### PR TITLE
utils/genio.py: added method for pattern match

### PR DESCRIPTION
**Pull request by rohan-r-deshpande**
- added a method read_line_with_matching_pattern()
- This method takes a file name, searches the file for the occurrences of specified "pattern".
- If found returns the line/lines in the form of a list.

Signed-off-by: Rohan Deshpande <rohan_d@linux.ibm.com>

### CODE REVIEW

File: avocado/utils/genio.py [hunk 1]
```diff
@@ -10,7 +10,12 @@
 #
 # This code was inspired in the autotest project,
 # client/shared/utils.py
-# Authors: Martin J Bligh <mbligh@google.com>, Andy Whitcroft <apw@shadowen.org>
+# Authors: Martin J Bligh <mbligh@google.com>
+#          Andy Whitcroft <apw@shadowen.org>
+#
+# Modification History:
+# Rohan Deshpande <rohan_d@linux.ibm.com>
+# - Added read_line_with_matching_pattern()
```
Comment (PraveenPenguin) [inline avocado/utils/genio.py:18]: as it is small addition so i feel we can drop that and we can get git history what added

**Review by PraveenPenguin — CHANGES_REQUESTED** [review]

File: avocado/utils/genio.py (hunk 1 above)
Comment (rohan-r-deshpande) [reply avocado/utils/genio.py:17]: Done

**Review by abdhaleegit — COMMENTED** [review]
@rohan-r-deshpande what if the none of the lines matches the pattern, should we handle that

**Comment by rohan-r-deshpande** [comment]
> @rohan-r-deshpande what if the none of the lines matches the pattern, should we handle that

@abdhaleegit Good point. 

Yes we **should** handle it.

As per the current code flow,  in case of no-match-found for the said patterns, eventually we would be returning an empty list to the caller.  So in my opinion, instead of handling this scenario here, we could might as well incorporate the same from the place where this function is being referenced. 

Reason for this approach?
 - We can neither raise an exception, nor fail it here, for not having the desired pattern found in the file.

Please let me know your opinion!

**Comment by abdhaleegit** [comment]
> > @rohan-r-deshpande what if the none of the lines matches the pattern, should we handle that
> 
> @abdhaleegit Good point.
> 
> Yes we **should** handle it.
> 
> As per the current code flow, in case of no-match-found for the said patterns, eventually we would be returning an empty list to the caller. So in my opinion, instead of handling this scenario here, we could might as well incorporate the same from the place where this function is being referenced.
> 
> Reason for this approach?
> 
> * We can neither raise an exception, nor fail it here, for not having the desired pattern found in the file.
> 
> Please let me know your opinion!

Yes, I agree . it is always good to handle exception at the calling program

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
@rohan-r-deshpande while reading content of a file, better to have it encoded `encoding="utf-8"`  the open call provides this option to encode

**Comment by rohan-r-deshpande** [comment]
> > > @rohan-r-deshpande what if the none of the lines matches the pattern, should we handle that
> > 
> > 
> > @abdhaleegit Good point.
> > Yes we **should** handle it.
> > As per the current code flow, in case of no-match-found for the said patterns, eventually we would be returning an empty list to the caller. So in my opinion, instead of handling this scenario here, we could might as well incorporate the same from the place where this function is being referenced.
> > Reason for this approach?
> > 
> > * We can neither raise an exception, nor fail it here, for not having the desired pattern found in the file.
> > 
> > Please let me know your opinion!
> 
> Yes, I agree . it is always good to handle exception at the calling program

Great! So, without changing anything further here, will handle the exceptions at function calls. Thanks!

**Comment by rohan-r-deshpande** [comment]
> > @rohan-r-deshpande while reading/writing content to a file, better to have it encoded `encoding="utf-8"` the open call provides this option to encode
> 
> Noted, will do the changes.

@abdhaleegit Done and re-submitted the patch. Thanks!

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**